editor/plugins/org.fusesource.ide.wsdl2rest.ui/src/org/fusesource/ide/wsdl2rest/ui/wizard/Wsdl2RestWizard.java

```diff
@@ -0,0 +1,359 @@
+				Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
+
+				Path outJavaPath = javaFile.toPath();
+				Wsdl2Rest tool = new Wsdl2Rest(wsdlLocation, outJavaPath);
+				initContextForTool(isBlueprint, isSpringBoot, camelFile, tool);
+				if (!Strings.isEmpty(options.getTargetServiceAddress())) {
+					URI targetAddressURI = new URI(options.getTargetServiceAddress());
+					URL targetAddressURL = targetAddressURI.toURL();
+					tool.setJaxwsAddress(targetAddressURL);
+				}
+				if (!Strings.isEmpty(options.getTargetRestServiceAddress())) {
+					URL targetRestAddressURL = new URL(options.getTargetRestServiceAddress());
+					tool.setJaxrsAddress(targetRestAddressURL);
+				}
+				if (outJavaPath != null && outJavaPath.toFile().exists()) {
+					tool.setJavaOut(outJavaPath);
+				}
+				ClassLoader loader = tool.getClass().getClassLoader();
+				Thread.currentThread().setContextClassLoader(loader);
+				tool.process();
+				updateDependencies();
+			}  finally {
+				Thread.currentThread().setContextClassLoader(oldLoader);
+				project.refreshLocal(IResource.DEPTH_INFINITE, new NullProgressMonitor());
+			}
+		}
+	}
+
+	/**
+	 * @param isBlueprint
+	 * @param isSpringBoot
+	 * @return
+	 */
+	public String getCamelFileName(boolean isBlueprint, boolean isSpringBoot) {
+		if (isBlueprint) {
+			return DEFAULT_BLUEPRINT_CONFIG_NAME;
+		}
+		if (isSpringBoot) {
+			return DEFAULT_SPRINGBOOT_CONFIG_NAME;
+		}
+		return DEFAULT_CONFIG_NAME;
+	}
+	
+	/**
+	 * @param isBlueprint
+	 * @param camelFile
+	 * @param tool
+	 */
+	protected Path initContextForTool(boolean isBlueprint, boolean isSpringBoot, File camelFile, Wsdl2Rest tool) {
+		Path contextpath = null;
+		if (camelFile != null && !camelFile.isDirectory()) {
+			contextpath = new File(camelFile.getAbsolutePath()).toPath();
+			tool.setCamelContext(contextpath);
+		} else {
+			String camelConfigName = getCamelFileName(isBlueprint, isSpringBoot);
+			if (camelFile != null) {
+				contextpath = new File(camelFile.getAbsolutePath() + File.separator + camelConfigName).toPath();
+				tool.setCamelContext(contextpath); 
+			} else {
+				IProject project = options.getProject();
+				IPath projectPath = ResourcesPlugin.getWorkspace().getRoot().findMember(project.getName()).getLocation();
+				contextpath = new File(projectPath.makeAbsolute() + File.separator + camelConfigName).toPath();
+				tool.setCamelContext(contextpath); 
+			}
+		}
+		return contextpath;
+	}
+
+	/**
+	 * @param resource
+	 * @return
+	 * @throws CoreException
+	 */
+	protected File findFileInEFS(IResource resource) throws CoreException {
+		File fileFound = null;
+
+		// gets URI for EFS.
+		URI uri = resource.getLocationURI();
+
+		// what if file is a link, resolve it.
+		if(resource.isLinked()){
+			uri = resource.getRawLocationURI();
+		}
+
+		// Gets native File using EFS
+		fileFound = EFS.getStore(uri).toLocalFile(0, new NullProgressMonitor());			
```

Comment:
optional: would be best to pass a progress monitor instead of using a NullProgressMonitor, ti will allow to provide a smoother progress reporting to user

Reply:
I created a more generic task for taking care of Progress Monitor https://issues.jboss.org/browse/FUSETOOLS-2968

Reply:
with a proposal to be tested https://github.com/apupier/jbosstools-fuse/tree/FUSETOOLS-2968-useProgressMonitor

Reply:
ok. something to explore once the first PR is in